substrate/frame/support/procedural/src/benchmark.rs

```diff
@@ -441,6 +441,37 @@ pub fn benchmarks(
 		benchmarks_by_name_mappings.push(quote!(#name_str => Self::#test_ident()))
 	}
 
+	let impl_test_function = content
+		.iter_mut()
+		.find_map(|item| {
+			let Item::Macro(item_macro) = item else {
+				return None;
+			};
+
+			if !item_macro
+				.mac
+				.path
+				.segments
+				.iter()
+				.any(|s| s.ident == "impl_benchmark_test_suite")
+			{
+				return None;
+			}
+
+			let tokens = item_macro.mac.tokens.clone();
+			*item = Item::Verbatim(quote! {});
```

Comment:
Just as note: You could also directly replace the `item`with the new quote code, but it looks cleaner like this.

Reply:
Actually, that's not possible, as it would be inserted before all the benchmarks structures and methods are defined, causing _undefined_ errors on checks (already tried that, actually 😅).